### PR TITLE
Fix timestep and other issues

### PR DESCRIPTION
**Pull request by bpbond**
This PR fixes some but not all issues currently causing R CMD CHECK to throw many errors and notes, including:

* `timestep` now passed as a formal parameter 
* `data-raw/` added to `.Rbuildignore`
* Many documentation problems fixed
* DESCRIPTION updated for R version dependency and `assertthat`
* Skeleton documentation added for `default_parameters`

### CODE REVIEW

File: R/default_parameters.R [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+#'
+#' Default parameters for SEM.
+#'
+#' @format A list with lots of entries:
+#' \describe{
+#'   \item{price}{price, in US dollars}
+#'   \item{carat}{weight of the diamond, in carats}
+#'   ...
+#' }
+"default_parameters"
```
Comment (kdorheim) [inline R/default_parameters.R:11]: Ah I thought that this sort of documentation went in `R/data.R` Oh but it looks like I made a mistake there, so let's delete that one or something.

**Review by kdorheim — APPROVED** [review]
Looks good just one comment to address. Thanks for making all the documentation changes, I copied and pasted the original documentation without checking to see if the any of the symbols caused issues with package check! 🙈

**Comment by bpbond** [comment]
@kdorheim Why are there copies of e.g. `SEM()` in both `model_functions.R` as well as `PestEcosysDemo.R`?

**Comment by kdorheim** [comment]
@bpbond ah! right so I am pretty sure that `PestEcosysDemo.R` is also not needed anymore... it was the original script from Deitz that we branched off of PestED. And then we turned it into a single function which wouldn't let users to set up their own runs

**Comment by bpbond** [comment]
Removed old code file @kdorheim

**Pull request merged**